packages/extension/src/components/pages/AddNewWallet/ImportNewWallet/ImportMnemonic/ImportMnemonic.tsx

```diff
@@ -17,18 +17,21 @@ export const ImportMnemonic: FC<ImportMnemonicProps> = ({ activeStep, password,
   const navigate = useNavigate();
   const { importMnemonic } = useWallet();
   const [mnemonicError, setMnemonicError] = useState('');
+  const mnemonicRef = useRef<HTMLInputElement | null>(null);
 
   const handleNext = useCallback(() => {
-    const isValidMnemonic = importMnemonic(
-      password,
-      (document.getElementById('mnemonic') as HTMLInputElement).value
-    );
-    if (isValidMnemonic) {
-      navigate(LIST_WALLETS);
-    } else if (isValidMnemonic === false) {
-      setMnemonicError('Your mnemonic is invalid');
+    let mnemonicValue = mnemonicRef.current?.value;
```

Comment:
I would replace this `let` by a `const` as you never re-assign that variable later on in the code.